src/sage/topology/moment_angle_complex.py

```diff
@@ -0,0 +1,781 @@
+            sage: M.components()
+            {(0, 1, 2): [The 2-simplex, The 2-simplex, The 2-simplex]}
+            sage: Z = MomentAngleComplex([[0], [1]]); Z
+            Moment-angle complex of Simplicial complex with vertex set (0, 1) and facets {(0,), (1,)}
+            sage: sorted(Z.components().items())
+            [((0,), [The 2-simplex, Minimal triangulation of the 1-sphere]),
+             ((1,), [Minimal triangulation of the 1-sphere, The 2-simplex])]
+
+        We interpret the output of this method by taking the product
+        of all the elements in each list, and then taking the union
+        of all products. From the previous example, we have
+        `\mathcal{Z} = S^1 \times D^2 \cup D^2 \times S^1 = \partial (D^2 \times D^2) = \partial D^4 = S^3`::
+
+            sage: Z = MomentAngleComplex([[0,1], [1,2], [2,3], [3,0]])
+            sage: sorted(Z.components().items())
+            [((0, 1),
+              [The 2-simplex,
+               The 2-simplex,
+               Minimal triangulation of the 1-sphere,
+               Minimal triangulation of the 1-sphere]),
+             ((0, 3),
+              [The 2-simplex,
+               Minimal triangulation of the 1-sphere,
+               Minimal triangulation of the 1-sphere,
+               The 2-simplex]),
+             ((1, 2),
+              [Minimal triangulation of the 1-sphere,
+               The 2-simplex,
+               The 2-simplex,
+               Minimal triangulation of the 1-sphere]),
+             ((2, 3),
+              [Minimal triangulation of the 1-sphere,
+               Minimal triangulation of the 1-sphere,
+               The 2-simplex,
+               The 2-simplex])]
+
+        It is not that difficult to prove that the previous moment-angle complex
+        is homeomorphic to a product of two 3-spheres. We can look at the
+        cohomologies to try and validate whether this makes sense::
+
+            sage: S3 = simplicial_complexes.Sphere(3)
+            sage: product_of_spheres = S3.product(S3)
+            sage: Z.cohomology()
+            {0: 0, 1: 0, 2: 0, 3: Z x Z, 4: 0, 5: 0, 6: Z}
+            sage: Z.cohomology() == product_of_spheres.cohomology()
+            True
+        """
+        return self._components
+
+    def dimension(self):
+        r"""
+        The dimension of ``self``.
+
+        The dimension of a moment-angle complex is the dimension
+        of the constructed (cubical) complex. It is not difficult to
+        see that this turns out to be `m+n+1`, where `m` is the number
+        of vertices and `n` is the dimension of the associated simplicial
+        complex.
+
+        EXAMPLES::
+
+            sage: Z = MomentAngleComplex([[0,1], [1,2,3]])
+            sage: Z.dimension()
+            7
+            sage: Z = MomentAngleComplex([[0, 1, 2]])
+            sage: Z.dimension()
+            6
+            sage: dim(Z)  # indirect doctest
```

Comment:
```suggestion
            sage: dim(Z)
```
This is not needed (since the method is called directly elsewhere).